src/OpenApi/src/OpenApiGenerator.cs

```diff
@@ -201,12 +201,40 @@ private static OpenApiResponses GetOpenApiResponses(MethodInfo method, EndpointM
                 responseContent[contentType] = new OpenApiMediaType();
             }
 
-            responses[statusCode] = new OpenApiResponse { Content = responseContent };
+            responses[statusCode] = new OpenApiResponse
+            {
+                Content = responseContent,
+                Description = GetResponseDescription(statusCode)
+            };
         }
-
         return responses;
     }
 
+    private static string GetResponseDescription(string statusCode)
+    {
+        if (statusCode.Length != 3)
+        {
+            return "";
+        }
+
+        char first = statusCode[0];
+        switch (first)
```

Comment:
@halter73's feedback earlier recommended that you use a switch expression here. There's a subtle difference between switch _statements_ (what the code is currently doing) and switch _expressions_ (the feedback given). Switch expressions are structured like `#value# switch => { #body# }`

```csharp
return first switch => {
  '1' => "Information";
  _ => string.Empty;
}
```